<a href="https://colab.research.google.com/github/luizrobertofreitas/colab-research-google/blob/master/RiscoDeUmAtivo.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [0]:
import numpy as np
import pandas as pd
from pandas_datareader import data as wb
import matplotlib.pyplot as plt

In [0]:
tickers = ['PG', 'BEI.DE']
sec_data = pd.DataFrame()
for t in tickers:
  sec_data[t] = wb.DataReader(t, data_source='yahoo', start='2007-1-1')['Adj Close']

In [4]:
sec_data.tail()

PG     BEI.DE
Date                             
2020-03-13  114.070000  84.800003
2020-03-16  108.500000  82.820000
2020-03-17  118.239998  85.839996
2020-03-18  117.449997  91.040001
2020-03-19  115.209999  90.419998

In [0]:
sec_returns = np.log(sec_data / sec_data.shift(1))

In [6]:
sec_returns

PG    BEI.DE
Date                          
2007-01-03       NaN       NaN
2007-01-04 -0.007621  0.006545
2007-01-05 -0.008624 -0.020772
2007-01-08  0.002202  0.000202
2007-01-09 -0.002517 -0.022858
...              ...       ...
2020-03-13  0.113409  0.003307
2020-03-16 -0.050062 -0.023626
2020-03-17  0.085966  0.035815
2020-03-18 -0.006704  0.058814
2020-03-19 -0.019256 -0.006834

[3326 rows x 2 columns]

# PG

In [7]:
sec_returns['PG'].mean()

0.00029337880647794634

In [8]:
sec_returns['PG'].mean() * 250

0.07334470161948659

In [9]:
sec_returns['PG'].std()

0.011584116095178329

In [10]:
sec_returns['PG'].std() * 250 ** 0.5 # potency of 1/2

0.18316095770289695

# Beiersdorf

In [11]:
sec_returns['BEI.DE'].mean()

0.00023114048957055425

In [12]:
sec_returns['BEI.DE'].mean() * 250

0.05778512239263856

In [13]:
sec_returns['BEI.DE'].std()

0.01368993169857746

In [15]:
sec_returns['BEI.DE'].std() * 250 ** 0.5 # potency of 1/2

0.21645682589821227

# Comparing

In [16]:
print("PG: " + str(sec_returns['PG'].mean() * 250))
print("BEI.DE: " + str(sec_returns['BEI.DE'].mean() * 250))

PG: 0.07334470161948659
BEI.DE: 0.05778512239263856


# Elegante Way

In [18]:
sec_returns[['PG', 'BEI.DE']].mean() * 250

PG        0.073345
BEI.DE    0.057785
dtype: float64

In [19]:
sec_returns[['PG', 'BEI.DE']].std() * 250 ** 0.5

PG        0.183161
BEI.DE    0.216457
dtype: float64

# Explicação

1. A média de retornos da PG neste momento é maior: 7% ao ano, apresentando menor volatilidade: 18%
2. A Beiersdorf apresenta maior risco, sendo que sua volatilidade é maior: 21% porém, seu retorno, em determinados momentos pode ser maior também!

